src/Board.java

```diff
@@ -13,6 +13,15 @@ public Board() {
         this.numberOfWhitePieces = 12;
     }
 
+    public Board(Board otherBoard) {
+        this.gameState = new ArrayList<Square>();
+        for (Square square : otherBoard.getGameState()) {
+            this.gameState.add(new Square(square));
+        }
```

Comment:
I was hoping there was a better way of making a deep copy of this than iterating through the `gameState` array, but upon quick investigation, it doesn't seem there is.